### ISSUE TITLE
Test Explorer should not ignore file excluded in .vscode/settings.json

### ISSUE THREAD

**Issue by kingsimba** [hbenl/vscode-mocha-test-adapter#103]
# Problem

I use TypeScript for coding and use "tsc --watch" to compile them to JS for mocha testing.
Because JS is much faster to run.

But when I exclude files in .vscode/settings.json, the test explorer won't be able to find any test.

```jsonc
{
    "files.exclude": {
        "node_modules": true,
        "test": true, // BUG: if ignored, there will be no test in Test Explorer
        ".gitignore": true
    },
    "cSpell.words": [
        "ETAG",
        "rpet"
    ],
    "mochaExplorer.files": "test/**/*.spec.js", // With this, the tests are still ignored.
    "mochaExplorer.require": "source-map-support/register"
}
```

# Suggestion

I think Test Explorer should obey "mochaExplorer.files" in settings.json and ignore "files.exclude".

**Comment by robertrossmann** [comment]
I use the same setup: I always pre-compile TypeScript sources to native JS files capable of being run directly by Node.js. I use `source-map-support` to map stack traces back to original source files so error tracking works well.

However, I always hide these compiled JS files from VS Code's view as I always create these JS files right next to their TypeScript sources. But since these files are hidden, Mocha Explorer just ignores them.

**Comment by kingsimba** [comment]
@robertrossmann Thanks. And I found another related inconvenience. Do you have any suggestions?

Because 'tsc --watch' is set up, when I press Ctrl+S to save a .spec.ts file, the corresponding .spec.js will be compiled. But the test explorer will not refresh to reload the JS file. (I have to click the refresh button)

**Comment by robertrossmann** [comment]
Yeah, well... I have bound a keyboard shortcut to refresh the tests. 🤷‍♂️😇

**Comment by kingsimba** [comment]
> Yeah, well... I have bound a keyboard shortcut to refresh the tests. 🤷‍♂️😇

But with TS, there is no such problem. So the problem is confirmed. I think it should be addresses as well. I shall open another issue on this. 🤝

**Comment by hbenl** [comment]
Fixed in version 2.4.0.
For now you still have to manually reload the tests after changing the sources - I will add a file watcher to fix this in one of the next releases.
